common/config/loader.go

```diff
@@ -90,14 +93,28 @@ func Load(env string, configDir string, zone string, config interface{}) error {
 	// TODO: remove log dependency.
 	stdlog.Printf("Loading config files=%v\n", files)
 
+	templateFuncs := sprig.FuncMap()
+
 	for _, f := range files {
 		// This is tagged nosec because the file names being read are for config files that are not user supplied
 		// #nosec
 		data, err := os.ReadFile(f)
 		if err != nil {
 			return err
 		}
-		err = yaml.Unmarshal(data, config)
+
+		tpl, err := template.New("config").Funcs(templateFuncs).Parse(string(data))
```

Comment:
is this compatible with the existing `config_template.yaml`? it looks like sprig provides a function used like `env "VAR"` but the template is written with things like `.Env.VAR`. do we have to migrate the template?

Reply:
The template would need adjusting. Sprig's `default` argument ordering is reversed from the `dockerize` version. There are probably a few other gotchas. I would suggest leaving the existing template as it is and creating a new template under `config/docker.yaml` with the updated syntax so that containers using dockerize will continue to work until we deprecate that. Newer containers can use this new functionality with the updated config file. I'll add a working config file to this PR.